components/tooltip/base.ts

```diff
@@ -425,6 +432,10 @@ export abstract class NzTooltipBaseComponent implements OnDestroy, OnInit {
   }
 
   hide(): void {
+    if (!this.nzOverlayClickable) {
```

Comment:
There are two ways that invoke `hide()` method:

Take nz-popconfirm for an example,

1. click outside overlay
2. overlay detachments

```html
<ng-template
      #overlay="cdkConnectedOverlay"
      cdkConnectedOverlay
      nzConnectedOverlay
      [cdkConnectedOverlayHasBackdrop]="nzBackdrop"
      [cdkConnectedOverlayOrigin]="origin"
      (overlayOutsideClick)="onClickOutside($event)"
      (detach)="hide()"
      (positionChange)="onPositionChange($event)"
      [cdkConnectedOverlayPositions]="_positions"
      [cdkConnectedOverlayOpen]="_visible"
      [cdkConnectedOverlayPush]="cdkConnectedOverlayPush"
      [nzArrowPointAtCenter]="nzArrowPointAtCenter"
    >
...
</ng-template>
```

If skip hiding the popconfirm here, the overlay detachments will not hide the popconfirm correctly.

So what about adding this condition at the `onClickOutside` method?

```diff
onClickOutside(event: MouseEvent): void {
+	if (!this.nzOverlayClickable) {
+		return;
+	}

  const target = _getEventTarget(event);
  if (!this.origin.nativeElement.contains(target) && this.nzTrigger !== null) {
    this.hide();
  }
}
```

Reply:
Yes, the interaction is wrong, I've done a bypass of the popconfirm internal button method here